man/pcb2gcode.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.49.3.
.TH PCB2GCODE "1" "November 2023" "pcb2gcode 2.5.0" "User Commands"
.SH NAME
pcb2gcode \- manual page for pcb2gcode 2.5.0
.SH DESCRIPTION
This manual page documents briefly the \fBpcb2gcode\fP command.
.PP
\fBpcb2gcode\fP is a program that takes the files typically produced by PCB
(printed circuit board) designing tools, in particular Gerber (RS-274X) and
Excellon (an RS-274C variant) files as parsed by \fBgerbv\fP, and produces
files that can be used to control a CNC (computer numerical control) milling
machine. These files are in G code format (RS-274D/NGC) that can be read, for
example, by the linuxcnc EMC2 system.

When these files are executed, an engraving bit is used to remove the surface
from a copper covered circuit board, thus creating isolated areas. In another
step, holes are drilled into the board at the appropriate positions, and it can
be cut out in another step. (For each step, a separate output file is created.)

These programs follow the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options is included below.

Instead of passing all options on the command line, nearly all options can be
stored in a file named \fImillproject\fP. There, options are given one per line
as \fIoption\fP=\fIvalue\fP pairs (without the two dashes), and can be
separated by blank lines and commented (everything after a `#` character is
treated as a comment). Options that don't take an argument (like
\fB\-\-metric\fP) are entered as \fIoption\fP=true or \fIoption\fP=1 there.

Unless configured otherwise, numeric values are in units of inch and
inch/minute. When the \fB\-\-metric\fP option is given, they are in mm and
mm/minute.
.PP
pcb2gcode 2.5.0
.SS "command line only options:"
.HP
\fB\-\-noconfigfile\fR [=arg(=1)] (=0) ignore any configuration file
.TP
\fB\-\-config\fR arg (=millproject)
list of comma\-separated config files
.TP
\-? [ \fB\-\-help\fR ]
produce help message
.TP
\fB\-V\fR [ \fB\-\-version\fR ]
show the current software version
.SS "Generic options (CLI and config files):"
.TP
\fB\-\-ignore\-warnings\fR [=arg(=1)] (=0)
Ignore warnings
.TP
\fB\-\-svg\fR arg
[DEPRECATED] use \fB\-\-vectorial\fR, SVGs will
be generated automatically; this option
has no effect
.TP
\fB\-\-metric\fR [=arg(=1)] (=0)
use metric units for parameters. does
not affect gcode output
.TP
\fB\-\-metricoutput\fR [=arg(=1)] (=0)
use metric units for output
.TP
\fB\-\-g64\fR arg
[DEPRECATED, use tolerance instead]
maximum deviation from toolpath,
overrides internal calculation
.TP
\fB\-\-tolerance\fR arg
maximum toolpath tolerance
.TP
\fB\-\-nog64\fR [=arg(=1)] (=0)
do not set an explicit g64
.TP
\fB\-\-output\-dir\fR arg
output directory
.TP
\fB\-\-basename\fR arg
prefix for default output file names
.TP
\fB\-\-preamble\-text\fR arg
preamble text file, inserted at the
very beginning as a comment.
.TP
\fB\-\-preamble\fR arg
gcode preamble file, inserted at the
very beginning.
.TP
\fB\-\-postamble\fR arg
gcode postamble file, inserted before
M9 and M2.
.TP
\fB\-\-no\-export\fR [=arg(=1)] (=0)
skip the exporting process
.SS "Drilling options, for making holes in the PCB:"
.TP
\fB\-\-drill\fR arg
Excellon drill file
.TP
\fB\-\-milldrill\fR [=arg(=1)] (=0)
[DEPRECATED] Use min\-milldrill\-hole\-dia
meter=0 instead
.TP
\fB\-\-milldrill\-diameter\fR arg
diameter of the end mill used for
drilling with \fB\-\-milldrill\fR
.TP
\fB\-\-min\-milldrill\-hole\-diameter\fR arg (=inf)
minimum hole width or milldrilling.
Holes smaller than this are drilled.
This implies milldrill
.TP
\fB\-\-zdrill\fR arg
drilling depth
.TP
\fB\-\-zmilldrill\fR arg
milldrilling depth
.TP
\fB\-\-drill\-feed\fR arg
drill feed in [i/m] or [mm/m]
.TP
\fB\-\-drill\-speed\fR arg
spindle rpm when drilling
.TP
\fB\-\-drill\-front\fR [=arg(=1)]
[DEPRECATED, use drill\-side instead]
drill through the front side of board
.TP
\fB\-\-drill\-side\fR arg (=auto)
drill side; valid choices are front,
back or auto (default)
.TP
\fB\-\-drills\-available\fR arg
list of drills available
.TP
\fB\-\-onedrill\fR [=arg(=1)] (=0)
use only one drill bit size
.TP
\fB\-\-drill\-output\fR arg (=drill.ngc)
output file for drilling
.TP
\fB\-\-nog91\-1\fR [=arg(=1)] (=0)
do not explicitly set G91.1 in drill
headers
.TP
\fB\-\-nog81\fR [=arg(=1)] (=0)
replace G81 with G0+G1
.TP
\fB\-\-nom6\fR [=arg(=1)] (=0)
do not emit M6 on tool changes
.TP
\fB\-\-milldrill\-output\fR arg (=milldrill.ngc)
output file for milldrilling
.SS "Milling options, for milling traces into the PCB:"
.TP
\fB\-\-front\fR arg
front side RS274\-X .gbr
.TP
\fB\-\-back\fR arg
back side RS274\-X .gbr
.TP
\fB\-\-voronoi\fR [=arg(=1)] (=0)
generate voronoi regions
.TP
\fB\-\-offset\fR arg (=0)
Note: Prefer to use \fB\-\-mill\-diameters\fR
and \fB\-\-milling\-overlap\fR if you just
that's what you mean.  An optional
offset to add to all traces, useful if
the bit has a little slop that you want
to keep out of the trace.
.TP
\fB\-\-mill\-diameters\fR arg (=0)
Diameters of mill bits, used in the
order that they are provided.
.TP
\fB\-\-milling\-overlap\fR arg (=50 %)
How much to overlap milling passes,
from 0% to 100% or an absolute length
.TP
\fB\-\-isolation\-width\fR arg (=0)
Minimum isolation width between copper
surfaces
.TP
\fB\-\-extra\-passes\fR arg (=0)
[DEPRECATED] use \fB\-\-isolation\-width\fR
instead. Specify the the number of
extra isolation passes, increasing the
isolation width half the tool diameter
with each pass
.TP
\fB\-\-pre\-milling\-gcode\fR arg
custom gcode inserted before the start
of milling each trace (used to activate
pump or fan or laser connected to fan)
.TP
\fB\-\-post\-milling\-gcode\fR arg
custom gcode inserted after the end of
milling each trace (used to deactivate
pump or fan or laser connected to fan)
.TP
\fB\-\-zwork\fR arg
milling depth in inches (Z\-coordinate
while engraving)
.TP
\fB\-\-mill\-feed\fR arg
feed while isolating in [i/m] or [mm/m]
.TP
\fB\-\-mill\-vertfeed\fR arg
vertical feed while isolating in [i/m]
or [mm/m]
.TP
\fB\-\-mill\-infeed\fR arg
maximum milling depth; PCB may be cut
in multiple passes
.TP
\fB\-\-mill\-speed\fR arg
spindle rpm when milling
.TP
\fB\-\-mill\-feed\-direction\fR arg (=0)
In which direction should all milling
occur
.TP
\fB\-\-invert\-gerbers\fR [=arg(=1)] (=0)
Invert polarity of front and back
gerbers, causing the milling to occur
inside the shapes
.TP
\fB\-\-draw\-gerber\-lines\fR [=arg(=1)] (=0)
Draw lines in the gerber file as just
lines and not as filled in shapes
.TP
\fB\-\-preserve\-thermal\-reliefs\fR [=arg(=1)] (=1)
generate mill paths for thermal reliefs
in voronoi mode
.TP
\fB\-\-front\-output\fR arg (=front.ngc)
output file for front layer
.TP
\fB\-\-back\-output\fR arg (=back.ngc)
output file for back layer
.SS "Outline options, for cutting the PCB out of the FR4:"
.TP
\fB\-\-outline\fR arg
pcb outline polygon RS274\-X .gbr
.TP
\fB\-\-fill\-outline\fR [=arg(=1)] (=1)
accept a contour instead of a polygon
as outline (enabled by default)
.TP
\fB\-\-cutter\-diameter\fR arg
diameter of the end mill used for
cutting out the PCB
.TP
\fB\-\-zcut\fR arg
PCB cutting depth in inches
.TP
\fB\-\-cut\-feed\fR arg
PCB cutting feed in [i/m] or [mm/m]
.TP
\fB\-\-cut\-vertfeed\fR arg
PCB vertical cutting feed in [i/m] or
[mm/m]
.TP
\fB\-\-cut\-speed\fR arg
spindle rpm when cutting
.TP
\fB\-\-cut\-infeed\fR arg
maximum cutting depth; PCB may be cut
in multiple passes
.TP
\fB\-\-cut\-front\fR [=arg(=1)]
[DEPRECATED, use cut\-side instead] cut
from front side.
.TP
\fB\-\-cut\-side\fR arg (=auto)
cut side; valid choices are front, back
or auto (default)
.TP
\fB\-\-bridges\fR arg (=0)
add bridges with the given width to the
outline cut
.TP
\fB\-\-bridgesnum\fR arg (=2)
specify how many bridges should be
created
.TP
\fB\-\-zbridges\fR arg
bridges height (Z\-coordinates while
engraving bridges, default to zsafe)
.TP
\fB\-\-outline\-output\fR arg (=outline.ngc)
output file for outline
.SS "Optimization options, for faster PCB creation, smaller output files, and different algorithms.:"
.TP
\fB\-\-optimise\fR [=arg(=2.54e\-06 m)] (=2.54e\-06 m)
Reduce output file size by up to 40%
while accepting a little loss of
precision.  Larger values reduce file
sizes and processing time even further.
.IP
Set to 0 to disable.
.TP
\fB\-\-eulerian\-paths\fR [=arg(=1)] (=1)
Don't mill the same path twice if
milling loops overlap.  This can save
up to 50% of milling time.  Enabled by
default.
.TP
\fB\-\-vectorial\fR [=arg(=1)] (=1)
enable or disable the vectorial
rendering engine
.TP
\fB\-\-tsp\-2opt\fR [=arg(=1)] (=1)
use TSP 2OPT to find a faster toolpath
(but slows down gcode generation)
.TP
\fB\-\-path\-finding\-limit\fR arg (=1)
Use path finding for up to this many
steps in the search (more is slower but
makes a faster gcode path)
.TP
\fB\-\-g0\-vertical\-speed\fR arg (=0.0211667 m s^\-1)
speed of vertical G0 movements, for use
in path\-finding
.TP
\fB\-\-g0\-horizontal\-speed\fR arg (=0.0423333 m s^\-1)
speed of horizontal G0 movements, for
use in path\-finding
.TP
\fB\-\-backtrack\fR arg (=inf)
allow retracing a milled path if it's
faster than retract\-move\-lower.  For
example, set to 5in/s if you are
willing to remill 5 inches of trace in
order to save 1 second of milling time.
.SS "Autolevelling options, for generating gcode to automatically probe the board and adjust milling depth to the actual board height:"
.TP
\fB\-\-al\-front\fR [=arg(=1)] (=0)
enable the z autoleveller for the front
layer
.TP
\fB\-\-al\-back\fR [=arg(=1)] (=0)
enable the z autoleveller for the back
layer
.TP
\fB\-\-software\fR arg
choose the destination software (useful
only with the autoleveller). Supported
programs are linuxcnc, mach3, mach4 and
custom
.TP
\fB\-\-al\-x\fR arg
max x distance between probes
.TP
\fB\-\-al\-y\fR arg
max y distance bewteen probes
.TP
\fB\-\-al\-probefeed\fR arg
speed during the probing
.TP
\fB\-\-al\-probe\-on\fR arg (=(MSG, Attach the probe tool)@M0 ( Temporary machine stop. ))
execute this commands to enable the
probe tool (default is M0)
.TP
\fB\-\-al\-probe\-off\fR arg (=(MSG, Detach the probe tool)@M0 ( Temporary machine stop. ))
execute this commands to disable the
probe tool (default is M0)
.TP
\fB\-\-al\-probecode\fR arg (=G31)
custom probe code (default is G31)
.TP
\fB\-\-al\-probevar\fR arg (=2002)
number of the variable where the result
of the probing is saved (default is
2002)
.TP
\fB\-\-al\-setzzero\fR arg (=G92 Z0)
gcode for setting the actual position
as zero (default is G92 Z0)
.SS "Alignment options, useful for aligning the milling on opposite sides of the PCB:"
.TP
\fB\-\-x\-offset\fR arg (=0)
offset the origin in the x\-axis by this
length
.TP
\fB\-\-y\-offset\fR arg (=0)
offset the origin in the y\-axis by this
length
.TP
\fB\-\-zero\-start\fR [=arg(=1)] (=0)
set the starting point of the project
at (0,0)
.TP
\fB\-\-mirror\-absolute\fR [=arg(=1)] (=1)
[DEPRECATED, must always be true]
mirror back side along absolute zero
instead of board center
.TP
\fB\-\-mirror\-axis\fR arg (=0)
For two\-sided boards, the PCB needs to
be flipped along the axis x=VALUE
.TP
\fB\-\-mirror\-yaxis\fR arg (=0)
For two\-sided boards, the PCB needs to
be flipped along the y axis instead
.SS "CNC options, common to all the milling, drilling, and cutting:"
.TP
\fB\-\-zsafe\fR arg
safety height (Z\-coordinate during
rapid moves)
.TP
\fB\-\-spinup\-time\fR arg (=0.001 s)
time required to the spindle to reach
the correct speed
.TP
\fB\-\-spindown\-time\fR arg
time required to the spindle to return
to 0 rpm
.TP
\fB\-\-zchange\fR arg
tool changing height
.TP
\fB\-\-zchange\-absolute\fR [=arg(=1)] (=0)
use zchange as a machine coordinates
height (G53)
.TP
\fB\-\-tile\-x\fR arg (=1)
number of tiling columns. Default value
is 1
.TP
\fB\-\-tile\-y\fR arg (=1)
number of tiling rows. Default value is
1
.PP
Git commit:
Boost: 108300
Gerbv: 2.10.0
Geos: Not installed


NEWS


	This project is developed using git.
	We do not currently maintain any dedicated NEWS
	file for the release versions, please refer to
	the git history.

-- 2015 Jan 26  Nicola Corna <nicola@corna.info>


pcb2gcode.sh
PCB2GCODE_BASE=/home/saintel/pcb2gcode/git_20150418/pcb2gcode/
PCB2GCODE=$PCB2GCODE_BASE/pcb2gcode
PCB2GCODE_M4=$PCB2GCODE_BASE/pcb2gcode_m4
MILLPROJECT=millproject

if [ -a $MILLPROJECT ] ; then {
   $PCB2GCODE --m4=true
   

   if (grep -q front $MILLPROJECT); then {

      if (grep -q ^front-output $MILLPROJECT); then {
         FRONT_OUTPUT=`grep front-output millproject | sed -e 's/^[ ]*front-output[ ]*=[ ]*\([^ \t]\+\)/\1/'`
      }; else { 
         FRONT_OUTPUT='front' 
      }; fi;
      FRONT_OUTPUT_M4=$FRONT_OUTPUT.m4
      FRONT_OUTPUT_GBR=$FRONT_OUTPUT.gbr

      if [ -a $FRONT_OUTPUT_M4 ]; then {
         echo "Processing front file : $FRONT_FILE" 
         m4 $PCB2GCODE_M4/gerber.m4 $FRONT_OUTPUT_M4 > $FRONT_OUTPUT_GBR
      }; else {
         echo "Front template file not found : $FRONT_OUTPUT_M4" 
      }; fi;

   } fi;
   
   if (grep -q ^back $MILLPROJECT); then {

      if (grep -q back-output $MILLPROJECT); then {
         BACK_OUTPUT=`grep back-output millproject | sed -e 's/^[ ]*back-output[ ]*=[ ]*\([^ \t]\+\)/\1/'`
      }; else { 
         BACK_OUTPUT='back' 
      }; fi;
      BACK_OUTPUT_M4=$BACK_OUTPUT.m4
      BACK_OUTPUT_GBR=$BACK_OUTPUT.gbr

      if [ -a $BACK_OUTPUT_M4 ]; then {
         echo "Processing back file : $BACK_FILE" 
         m4 $PCB2GCODE_M4/gerber.m4 $BACK_OUTPUT_M4 > $BACK_OUTPUT_GBR
      }; else {
         echo "Back template file not found : $BACK_OUTPUT_M4" 
      }; fi;

   } fi;

   if (grep -q outline $MILLPROJECT); then {

      if (grep -q outline-output $MILLPROJECT); then {
         OUTLINE_OUTPUT=`grep outline-output millproject | sed -e 's/^[ ]*outline-output[ ]*=[ ]*\([^ \t]\+\)/\1/'`
      }; else { 
         OUTLINE_OUTPUT='outline' 
      }; fi;
      OUTLINE_OUTPUT_M4=$OUTLINE_OUTPUT.m4
      OUTLINE_OUTPUT_GBR=$OUTLINE_OUTPUT.gbr

      if [ -a $OUTLINE_OUTPUT_M4 ]; then {
         echo "Processing outline file : $OUTLINE_FILE" 
         m4 $PCB2GCODE_M4/gerber.m4 $OUTLINE_OUTPUT_M4 > $OUTLINE_OUTPUT_GBR
      }; else {
         echo "Outline template file not found : $OUTLINE_FILE_M4" 
      }; fi;

   } fi;
   
} fi;

